opsdroid/logging.py

```diff
@@ -60,71 +61,92 @@ def filter(self, record):
-        if config["logging"]["extended"]:
-            formatter_str = formatter_str[:-1] + ".%(funcName)s():"
+    if config.get("timestamp"):
+        formatter_str = "%(asctime)s " + formatter_str
 
     formatter_str += " %(message)s"
-    formatter = logging.Formatter(formatter_str)
 
-    console_handler = logging.StreamHandler()
-    console_handler.setLevel(log_level)
-    console_handler.setFormatter(formatter)
+    return formatter_str
 
-    with contextlib.suppress(KeyError):
-        console_handler.addFilter(ParsingFilter(config, config["logging"]["filter"]))
 
-    rootlogger.addHandler(console_handler)
+def configure_logging(config):
+    """Configure the root logger based on user config."""
+    rootlogger = logging.getLogger()
+    while rootlogger.handlers:
+        rootlogger.handlers.pop()
 
-    with contextlib.suppress(KeyError):
-        if not config["logging"]["console"]:
-            console_handler.setLevel(logging.CRITICAL)
+    try:
+        if config["path"]:
+            logfile_path = os.path.expanduser(config["path"])
+        else:
+            logfile_path = config["path"]
+    except KeyError:
+        logfile_path = DEFAULT_LOG_FILENAME
 
     if logfile_path:
         logdir = os.path.dirname(os.path.realpath(logfile_path))
         if not os.path.isdir(logdir):
             os.makedirs(logdir)
 
-        file_handler = RotatingFileHandler(
-            logfile_path, maxBytes=logging_config.get("file-size", 50e6)
+    log_level = get_logging_level(config.get("level", "info"))
+    rootlogger.setLevel(log_level)
+    formatter_str = set_formatter_string(config)
+    formatter = logging.Formatter(formatter_str)
+    handler = None
+
+    if config.get("rich") is not False:
+        handler = RichHandler(
+            rich_tracebacks=True,
+            show_time=config.get("timestamp", True),
+            show_path=config.get("extended", True),
         )
 
+    if logfile_path:
+        file_handler = RotatingFileHandler(
+            logfile_path, maxBytes=config.get("file-size", 50e6)
+        )
         file_handler.setLevel(log_level)
         file_handler.setFormatter(formatter)
+        rootlogger.addHandler(file_handler)
 
-        with contextlib.suppress(KeyError):
-            file_handler.addFilter(ParsingFilter(config, config["logging"]["filter"]))
+    if config.get("console"):
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+
+    # If we still don't have the handler, we are assuming that
+    # the user wants to switch off logging, let's log only
+    # Critical errors
+    if not handler:
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+        log_level = get_logging_level("critical")
+
+    if config.get("filter") and handler:
+        handler.addFilter(ParsingFilter(config, config["filter"]))
+    if handler:
+        handler.setLevel(log_level)
+        rootlogger.addHandler(handler)
```

Comment:
```suggestion
        rootlogger.addHandler(handler)
        if config.get("console") and config.get("rich"):
            _LOGGER.warning("Both console and rich logging are enabled. Using console logging because console takes precedence over rich logging.")

```

Reply:
I've tried to add a test for this and even though I see the log, when the test is run this is never shown both with caplog or capsys and I'd rather not block this PR with a failure of 100% coverage for this warning since logging would still work just not as pretty